src/main/java/com/dev/cinema/util/PasswordValidator.java

```diff
@@ -0,0 +1,31 @@
+package com.dev.cinema.util;
+
+import com.dev.cinema.annotaion.ValidPassword;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import org.springframework.beans.BeanWrapperImpl;
+
+public class PasswordValidator implements ConstraintValidator<ValidPassword, Object> {
+    private String field;
+    private String fieldMatch;
+
+    @Override
+    public void initialize(ValidPassword constraintAnnotation) {
+        this.field = constraintAnnotation.field();
+        this.fieldMatch = constraintAnnotation.fieldMatch();
+    }
+
+    @Override
+    public boolean isValid(Object user, ConstraintValidatorContext constraintValidatorContext) {
+        Object fieldValue = new BeanWrapperImpl(user)
+                .getPropertyValue(field);
+        Object fieldMatchValue = new BeanWrapperImpl(user)
+                .getPropertyValue(fieldMatch);
+
+        if (fieldValue != null) {
+            return fieldValue.equals(fieldMatchValue);
+        } else {
+            return fieldMatchValue == null;
```

Comment:
at this point if fieldValue == null and fieldMatchValue == null you will register your user